Assets/Scripts/GameState/LeaderboardData.cs

```diff
@@ -4,48 +4,67 @@
 using System.Runtime.Serialization.Formatters.Binary;
 
 [System.Serializable]
-public class Highscores {
-    public int[] score;
-    public string[] name;
+public struct Highscore {
+    public int score;
+    public string name;
 
-    public Highscores(int[] Score, string[] Name)
+    public Highscore(int Score, string Name)
     {
         score = Score;
         name = Name;
     }
 }
 
+class HighscoreComparator : IComparer<Highscore>
+{
+    public int Compare(Highscore x, Highscore y)
+    {
+        if (x.score == 0)
+        {
+            return y.score;
+        }
+        if(y.score == 0)
+        {
+            return x.score;
+        }
+
+        return y.score.CompareTo(x.score);
+    }
+
+}
+
 public static class LeaderboardData
 {
     // Path to where highscore data is saved
     private static string path = Application.persistentDataPath + "/highscores.data";
 
+    /// <summary>
+    /// Add a new score to the leaderboard
+    /// </summary>
+    /// <param name="score"></param>
+    /// <param name="name"></param>
     public static void AddScore(int score, string name)
     {
-        Highscores highscores = LoadScores();
-        Highscores newScores = highscores;
+        List<Highscore> highscores = LoadScores();
+        highscores.Add(new Highscore(score, name));
 
-        for(int i = 0; i < 10; i++)
-        {
-            if(highscores.score[i] < score)
-            {
-                newScores.score[i] = score;
-                newScores.name[i] = name;
+        // Sort the struct using a custom comparator
+        HighscoreComparator HC = new HighscoreComparator();
 
-                for(int n = i + 1; n < 10; n++)
-                {
-                    newScores.score[n] = highscores.score[n - 1];
-                    newScores.name[n] = highscores.name[n - 1];
-                }
+        highscores.Sort(HC);
 
-                break;
-            }
+        foreach(Highscore h in highscores)
+        {
+            Debug.Log(h.name + " : " + h.score.ToString());
```

Comment:
Use string interpolation instead of concatenating strings (this also removes the need to call objects' `ToString()` method): 🙂
```suggestion
            Debug.Log($"{h.name} : {h.score}");
```
That being said, logging statements (apart from logging errors or warnings) should generally be removed from production code, as I'm sure you're aware of 😅